### PR TITLE
CTCP-5892: Updated XSDs and resolved compilation issues

### PR DESCRIPTION
**Pull request by kevincoatsworth**

### CODE REVIEW

File: app/services/submission/SubmissionService.scala [hunk 1]
```diff
@@ -154,7 +148,7 @@ class SubmissionService @Inject() (
         case _                                            => placeOfLoading
       },
       HouseConsignment = houseConsignments match {
-        case Nil => Seq(HouseConsignmentType06("1", Nil))
+        case Nil => Seq(HouseConsignmentType06(BigInt("1"), Nil))
```
Comment (jackheslop96) [inline app/services/submission/SubmissionService.scala:151]: Think `BigInt("1")` can just be `1`. The compiler should be able to handle interpreting an Int as a BigInt I believe.

File: app/services/submission/SubmissionService.scala [hunk 2]
```diff
@@ -240,7 +234,7 @@ class SubmissionService @Inject() (
       identificationNumber <- (__ \ TransportMeansIdentificationNumberPage(index).toString).read[String]
       nationality          <- (__ \ TransportMeansNationalityPage(index).toString).read[Nationality]
     } yield DepartureTransportMeansType05(
-      sequenceNumber = index.sequenceNumber,
+      sequenceNumber = BigInt(index.sequenceNumber),
```
Comment (jackheslop96) [inline app/services/submission/SubmissionService.scala:237]: Is there any reason for the `sequenceNumber` val inside `Index` to remain a `String`? Could we just have `val sequenceNumber: BigInt = display`? Then there would be no need to wrap everything in a `BigInt` here.

File: app/services/submission/SubmissionService.scala [hunk 3]
```diff
@@ -257,7 +251,7 @@ class SubmissionService @Inject() (
       nationality                          <- (__ \ NationalityPage(index).toString).read[Nationality]
       conveyanceReferenceNumber            <- (__ \ ConveyanceReferenceNumberPage(index).toString).readNullable[String]
     } yield ActiveBorderTransportMeansType03(
-      sequenceNumber = index.sequenceNumber,
+      sequenceNumber = BigInt(index.sequenceNumber),
```
Comment (jackheslop96) [inline app/services/submission/SubmissionService.scala:254]: As above

File: app/services/submission/SubmissionService.scala [hunk 4]
```diff
@@ -274,18 +268,18 @@ class SubmissionService @Inject() (
     def sealReads(sealIndex: Index): Reads[SealType05] =
       (__ \ SealIdentificationNumberPage(equipmentIndex, sealIndex).toString)
         .read[String]
-        .map(SealType05(sealIndex.sequenceNumber, _))
+        .map(SealType05(BigInt(sealIndex.sequenceNumber), _))
```
Comment (jackheslop96) [inline app/services/submission/SubmissionService.scala:271]: As above

File: app/services/submission/SubmissionService.scala [hunk 5]
```diff
@@ -274,18 +268,18 @@ class SubmissionService @Inject() (
...
 
     def goodsReferenceReads(itemIndex: Index): Reads[GoodsReferenceType02] =
       __.read[Item]
         .map(_.declarationGoodsItemNumber)
-        .map(GoodsReferenceType02(itemIndex.sequenceNumber, _))
+        .map(GoodsReferenceType02(BigInt(itemIndex.sequenceNumber), _))
```
Comment (jackheslop96) [inline app/services/submission/SubmissionService.scala:276]: As above

File: app/services/submission/SubmissionService.scala [hunk 6]
```diff
@@ -274,18 +268,18 @@ class SubmissionService @Inject() (
...
 
     for {
       containerIdNo   <- (__ \ ContainerIdentificationNumberPage(equipmentIndex).toString).readNullable[String]
       seals           <- (__ \ SealsSection(equipmentIndex).toString).readArray[SealType05](sealReads)
       goodsReferences <- (__ \ ItemsSection(equipmentIndex).toString).readArray[GoodsReferenceType02](goodsReferenceReads)
-    } yield TransportEquipmentType06(equipmentIndex.sequenceNumber, containerIdNo, seals.length, seals, goodsReferences)
+    } yield TransportEquipmentType06(BigInt(equipmentIndex.sequenceNumber), containerIdNo, seals.length, seals, goodsReferences)
```
Comment (jackheslop96) [inline app/services/submission/SubmissionService.scala:282]: As above

File: app/services/submission/SubmissionService.scala [hunk 7]
```diff
@@ -309,7 +303,7 @@ class SubmissionService @Inject() (
         identificationNumber <- (__ \ IdentificationNumberPage(hcIndex, dtmIndex).toString).read[String]
         nationality          <- (__ \ CountryPage(hcIndex, dtmIndex).toString).read[Nationality]
       } yield DepartureTransportMeansType05(
-        sequenceNumber = dtmIndex.sequenceNumber,
+        sequenceNumber = BigInt(dtmIndex.sequenceNumber),
```
Comment (jackheslop96) [inline app/services/submission/SubmissionService.scala:306]: As above

File: app/services/submission/SubmissionService.scala [hunk 8]
```diff
@@ -319,7 +313,7 @@ class SubmissionService @Inject() (
     (__ \ DepartureTransportMeansListSection(hcIndex).toString).readArray[DepartureTransportMeansType05](departureTransportMeansReads).map {
       departureTransportMeans =>
         HouseConsignmentType06(
-          sequenceNumber = hcIndex.sequenceNumber,
+          sequenceNumber = BigInt(hcIndex.sequenceNumber),
```
Comment (jackheslop96) [inline app/services/submission/SubmissionService.scala:316]: As above

File: test/base/TestMessageData.scala [hunk 9]
```diff
@@ -157,7 +145,7 @@ trait TestMessageData {
       AdditionalSupplyChainActor = Nil,
       TransportEquipment = Seq(
         TransportEquipmentType06(
-          sequenceNumber = "1",
+          sequenceNumber = BigInt("1"),
```
Comment (jackheslop96) [inline test/base/TestMessageData.scala:148]: Can just be `1`

File: test/base/TestMessageData.scala [hunk 10]
```diff
@@ -183,7 +171,7 @@ trait TestMessageData {
       CountryOfRoutingOfConsignment = Nil,
       ActiveBorderTransportMeans = Seq(
         ActiveBorderTransportMeansType02(
-          sequenceNumber = "1",
+          sequenceNumber = BigInt("1"),
```
Comment (jackheslop96) [inline test/base/TestMessageData.scala:174]: Can just be `1`

File: test/navigator/EquipmentNavigatorSpec.scala [hunk 11]
```diff
@@ -48,7 +48,9 @@ class EquipmentNavigatorSpec extends SpecBase with ScalaCheckPropertyChecks with
 
         "to SealIdentifierNumberPage when user selects No and AuthType is in CL253" in {
           val userAnswers = emptyUserAnswers
-            .copy(departureData = basicIe015.copy(Authorisation = Seq(AuthorisationType03("1", ACR, "1234"), AuthorisationType03("2", SSE, "1235"))))
+            .copy(departureData =
+              basicIe015.copy(Authorisation = Seq(AuthorisationType03(BigInt(1), ACR, "1234"), AuthorisationType03(BigInt(2), SSE, "1235")))
```
Comment (jackheslop96) [inline test/navigator/EquipmentNavigatorSpec.scala:52]: Can just be `1`

File: test/navigator/EquipmentNavigatorSpec.scala [hunk 12]
```diff
@@ -88,7 +90,7 @@ class EquipmentNavigatorSpec extends SpecBase with ScalaCheckPropertyChecks with
           "when ContainerIndicatorPage is true" - {
             "must navigate to AddContainerIdentificationNumberYesNoPage " in {
               val userAnswers = emptyUserAnswers
-                .copy(departureData = basicIe015.copy(Authorisation = Seq(AuthorisationType03("1", ACR, "test"))))
+                .copy(departureData = basicIe015.copy(Authorisation = Seq(AuthorisationType03(BigInt(1), ACR, "test"))))
```
Comment (jackheslop96) [inline test/navigator/EquipmentNavigatorSpec.scala:93]: Can just be `1`

**Comment by kevincoatsworth** [comment]
rebuild

**Review by jackheslop96 — APPROVED** [review]

**Pull request merged**